### PR TITLE
Feat/dashboard swap

### PR DESCRIPTION
**Pull request by shikhar360**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/contracts/contracts/compound/CTokenFirstExtension.sol [hunk 1]
```diff
@@ -657,6 +638,35 @@ contract CTokenFirstExtension is
     return balance;
   }
 
+  function swapCollateral() public {
```
Comment (vminkov) [inline packages/contracts/contracts/compound/CTokenFirstExtension.sol:641]: add these public fns to the `_getExtensionFunctions` list

File: packages/contracts/contracts/compound/CTokenFirstExtension.sol [hunk 2]
```diff
@@ -657,6 +638,35 @@ contract CTokenFirstExtension is
...
+    uint256 underlyingBalance = asCToken().balanceOfUnderlying(msg.sender);
+    asCToken().flash(underlyingBalance, new bytes(0));
+  }
+
+  function receiveFlashLoan(uint256 amount, bytes calldata data) public {
+    (address borrower, address outputToken, address newCollateralMarket, address target, bytes memory swapData) = abi
+      .decode(data, (address, address, address, address, bytes));
```
Comment (vminkov) [inline packages/contracts/contracts/compound/CTokenFirstExtension.sol:648]: actually, is this fn even used or dead code that is to be removed?

File: packages/contracts/contracts/ionic/CollateralSwap.sol [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+import { IERC20, SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+
+import { IFlashLoanReceiver } from "./IFlashLoanReceiver.sol";
+import { Exponential } from "../compound/Exponential.sol";
+import { ICErc20 } from "../compound/CTokenInterfaces.sol";
+import { IonicComptroller } from "../compound/ComptrollerInterface.sol";
+
+// import { console } from "forge-std/console.sol";
```
Comment (vminkov) [inline packages/contracts/contracts/ionic/CollateralSwap.sol:12]: let's merge my other PR here?

File: packages/contracts/contracts/compound/CTokenFirstExtension.sol (hunk 2 above)
Comment (rhlsthrm) [reply packages/contracts/contracts/compound/CTokenFirstExtension.sol]: Correct, this def needs to be removed.

**Pull request merged**